src/Extractor/DB2ResultWriter.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\DbExtractor\Extractor;
+
+use Iterator;
+use Keboola\Csv\CsvWriter;
+use Keboola\DbExtractor\Adapter\ResultWriter\DefaultResultWriter;
+use Keboola\DbExtractor\Adapter\ValueObject\QueryMetadata;
+use Keboola\DbExtractorConfig\Configuration\ValueObject\ExportConfig;
+
+class DB2ResultWriter extends DefaultResultWriter
+{
+    protected function writeRows(
+        Iterator $iterator,
+        QueryMetadata $queryMetadata,
+        ExportConfig $exportConfig,
+        CsvWriter $csvWriter
+    ): void {
+        // Write the rest
+        $this->rowsCount = 0;
+        $this->lastRow = null;
+        while ($iterator->valid()) {
+            $resultRow = (array) $iterator->current();
+            // Write header
+            if ($this->rowsCount === 0 && $this->hasCsvHeader($exportConfig)) {
```

Comment:
tohle nevím jak jinak udělat... ale pokud bude prázdná tabulka tak to header nezapíše do csvčka

Reply:
Nie som si isty ci to pojde.

V tomto teste nemas asi CSV header ani `columns` v manifeste:
https://github.com/keboola/db-extractor-db2/tree/45536370a8b3255aec3516c5e261c6cd2008dd8b/tests/functional/run-query-no-rows/expected/data/out/tables

Reply:
Ako to bolo riesene v starom extractore?

Reply:
no právě že ve starým to bylo taky tak
https://github.com/keboola/db-extractor-db2-deprecated/blob/master/src/Keboola/DbExtractor/Extractor/DB2.php#L80

Reply:
> V tomto teste nemas asi CSV header ani columns v manifeste:

jj vím, ale nevím jak ty columns jinak dostat

Reply:
Co ak by sa pouzil `ODBC` adapter bez PDO?
https://github.com/keboola/db-extractor-adapter/tree/master/src/ODBC

Reply:
priklad: https://github.com/keboola/db-extractor-netsuite/tree/master/src/Extractor

Reply:
odbc asi ano ale na to upřímně už nemám kapacitu - už takhle jsem tím zabil 1/2 dne

napíšu vladovi a popíšu problém ať rozhodne, protože tím nestihnu jiné věci ze sprintu

Reply:
ja sa len bojim toho, ze o tyzden to tu mame zase s tym, ze nejde exportovat prazdnu tabulku

Reply:
skusim dat tomu odbc 30min

Reply:
no jestli to dáš za 30 (už jen 19 :)) minut tak si dobrej :)

Reply:
uz som to vymenil :slightly_smiling_face:  cakam kde to spadne v testoch

Reply:
(pozeram, ze je to nejake zaseknute :slightly_smiling_face: )
https://github.com/keboola/db-extractor-db2/blob/4ad6233ad8bc3156825a527971755228b77dc846/tests/phpunit/bootstrap.php#L7

Reply:
spojenie ide, skusim pofixovat testy

Reply:
@michaljurecko povedlo se?

Reply:
@ondrajodas este dorabam chybajucu podporu PK, vyzera to dobre, ked tam nebude nejake prekvapenie v produkcii, testy mam uz zelene